enhancements/cluster-logging/gcp-external-account-auth.md

```diff
@@ -0,0 +1,180 @@
+static keys in Secrets.  The ClusterLogForwarder is currently blocked in this effort due to lack of upstream 
+support for the feature in our collector. This enhancement would enable the feature in upstream Vector,
+allowing the ClusterLogForwarder to be configured to successfully authenticate by using WIF.
+
+### User Stories
+* As an OpenShift administrator, I want the ClusterLogForwarder to use a credential file of type 'external_account'
+to forward logs to GCP Cloud Logging service, utilizing WIF and removing the need for long-lived, static keys.
+* As an engineer, I want the Vector collector to authenticate using an 'external_account' credential file type 
+configured in the ClusterLogForwarder, successfully forwarding logs to GCP, while ensuring compatibility with 
+both service_account and external_account (WIF) credentials file types.
+
+### Goals
+* Modify the GCP module in upstream Vector to support 'external_account' credentials. 
+  * Get the PR approved and merged into upstream.
+* Implement the updated GCP module into the OpenShift build of Vector.
+* Enable the feature in ClusterLogForwarder for the GCP output so that collector can authenticate and 
+forward logs to GCP using WIF
+* Ensure compatibility with existing key-based GCP 'service_account' authentication
+
+### Non-Goals
+* Modifying the ClusterLogForwarder Spec or OpenShift Observability API Model. 
+* Supporting GCP authentication methods other than credentials files of type 'service_account' or 'external_account'.
+
+## Proposal
+Vector lacks an official SDK for their Rust Language and the current library is not well maintained. To 
+integrate WIF, Vector’s GCP authentication module will need to be extended to support 'external_account' 
+credential file type. 
+
+The Vector collector will:
+
+* Determine the authentication type by reading the credentials JSON file. 
+* Extract the OpenShift Service Account token from the local volume. 
+* Exchange the OpenShift token for a Google Identity Token using Google STS. 
+* Exchange the Identity Token for a short-lived Access Token via Google IAM impersonation. 
+* Use the resulting Access Token in the log-forwarding request to GCP services.
+
+The ClusterLogForwarder will:
+
+* Read the 'type' from the configured credentials file, specified in a secret by the user.
+* Conditionally project the service account token if the type is 'external_account'
+* Create the collector configuration in the same way regardless of type, pointing to the path of the secret
+as the 'credentials_path'.
+
+### Workflow Description
+### API Extensions
+No changes are necessary.  The existing `credentials` secretRef (key and secretName) can be used:
+```yaml
+ googleCloudLogging:
+   authentication:
+     credentials:
+       key: google-credentials-file.json
+       secretName: gcp-secret
+```
+Note: A configMap would be sufficient here since the 'external_account' file type contains no sensitive data,
```

Comment:
Nit.  We can drop this note given we are not changing the API and we have already stated we can use what exists.